Resources/public/js/views/services/plugins/ez-searchplugin.js

```diff
@@ -440,6 +536,55 @@ YUI.add('ez-searchplugin', function (Y) {
                 callback(err, locationStructArr);
             }, this));
         },
+
+        /**
+         * Loads Location for each content struct in given array and puts it in the new `location`
+         * field for the content struct.
+         *
+         * @method _loadLocationForContent
+         * @protected
+         * @param {Array|Null} contentStructArr
+         * @param {Function} callback
+         */
+        _loadLocationForContent: function (viewName, contentStructArr, callback) {
+            var contentService = this._getContentService(),
+                locationsIds,
+                query;
+
+            locationsIds = Y.Array.reduce(contentStructArr, '', function (previousId, struct, index) {
+                // TODO this is bad but no other way it seems
+                var locationId = struct.content.get('resources').MainLocation.split('/').pop();
```

Comment:
Hmm ignore, this is just theoretical possibility, so not something we need to spend time on now. UI and all other use of the product is currently always creating locations.